src/SimpleRSS.Persistence/Store/InMemory.fs

```diff
@@ -8,40 +8,53 @@ open SimpleRSS.Persistence.Types.Connector
 
 module InMemory =
 
+    type Item<'D> = int * 'D
+
     /// create a `Store` with a `mutable list` as the db.
     /// implements Connector.Store
     type InMemoryStore<'T>() =
 
-        let mutable db : 'T list = []
+        let mutable db : Item<'T> list = []
+
+        let nextId (d: (int * 'T) list) : int =
```

Comment:
This could be problematic. We could end up with duplicate ID's. If an item is created with ID 0, the caller will probably store that ID. If that item gets deleted and a new item gets added, it will again have ID 0. If the original caller then requests the item it added with the ID 0 it received, the returned entity will be something completely different from what it expects.
I suggest we just use an atomic counter or GUID's for generating ID's.

Reply:
Was specifically trying to avoid using an independent counter, and i also like the int based counter from a readability perspective - but we can use guids - i'm cool with that

Reply:
Out of curiosity, why are you trying to avoid a counter?

Reply:
Would like to minimize the number of mutable values, but other than that don't really mind (I mean the entire connector is mutable at its core anyway)